### PR TITLE
feat(hmi-client): fixed logging and added toast service

### PR DESCRIPTION
**Pull request by tangopapatime**
# Description

The vue logger didn't really work very well in our use case so I created our own.  I added a little toast system that can be activated globally when creating a new Logger:

```
export const logger = new Logger({
	consoleEnabled: !isProduction,
	callerInfoEnabled: true,
	**showToast: false**
});
```

By default, all logs will be buffered and sent to hmi-server but you can pass in an additional message option:

```
interface ILoggerMessageOptions {
	showToast?: boolean; // will force a show or surpression of toast
	toastTitle?: string; // change default title of toast
	silent?: boolean; // do not transmit to backend
}
```

logger.error(error, { showToast: true, toastTitle: `Authentication Error` });

The toast service can also be used independent of the logger:

```
import Toast from 'primevue/toast';
import { ToastSummaries, ToastSeverity, useToastService } from '@/services/toast';

const toast = useToastService();

toast.showToast(
			ToastSeverity.error,
			`${ToastSummaries.NETWORK_ERROR} (${status})`,
			'Unauthorized',
			5000
		);
```

Resolves #(issue)

### CODE REVIEW

**Comment by tangopapatime** [comment]
error

File: packages/client/hmi-client/src/api/api.ts [hunk 1]
```diff
@@ -20,7 +22,29 @@ API.interceptors.request.use(
 		return config;
 	},
 	(error) => {
-		console.error(error);
+		logger.error(error, { showToast: true, toastTitle: `Authentication Error` });
```
Comment (YohannParis) [inline packages/client/hmi-client/src/api/api.ts:25]: I don't think we want to do this here, we should do like on the `App.vue` and redirect the user to the login page.

Also can we merge those two interceptors (from `App.vue` and here)? I don't understand why we should have two.

File: packages/client/hmi-client/src/services/toast.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+export const useToastService = () => {
+	const showToast = (
+		severity: ToastSeverity,
+		summary: string,
+		detail: string,
+		life: number = 4000
+	) => {
+		ToastEventBus.emit('add', {
+			severity: severity,
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/toast.ts:31]: In JS now you don't need to explicitely write the value if it's the same name as the property.
i.e. `{ severety, summary, detail, life }` will do it.

File: packages/client/hmi-client/src/utils/log-buffer.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			config.headers.Authorization = `Bearer ${auth.token}`;
+		} else {
+			config.headers = { Authorization: `Bearer ${auth.token}` };
+		}
+		return config;
+	},
+	(error) => {
+		console.warn(error);
+	}
+);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/utils/log-buffer.ts:27]: We already have an Axios call API for that same purpose, and the same kind of intereceptors on the App.vue.
I think all of those needs to be refactored.

File: packages/client/hmi-client/src/utils/log-buffer.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+const toast = useToastService();
+
+// LogDetails Interface
+interface LogDetails {
```
Comment (YohannParis) [inline packages/client/hmi-client/src/utils/log-buffer.ts:32]: Now we end types with the word `Type`, [see documentation](https://github.com/DARPA-ASKEM/TERArium/blob/main/packages/client/hmi-client/DEVELOPMENT.md#types)

File: packages/client/hmi-client/src/utils/log-buffer.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	add = (log: LogDetails) => {
+		this.logs.push(log);
+	};
+
+	isEmpty = (): boolean => {
+		if (this.logs.length > 0) {
+			return false;
+		}
+		return true;
+	};
```
Comment (YohannParis) [inline packages/client/hmi-client/src/utils/log-buffer.ts:57]: We are using lodash in the app, so you can use their `isEmtpy()` instead

File: packages/client/hmi-client/src/utils/log-buffer.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+			try {
+				const resp = await LOGS.post(`/logs/`, {
+					logs: this.getLogBuffer()
+				});
+				const { status } = resp;
+				if (status === 200) {
+					this.clearLogs();
+				}
+			} catch (error: any) {
+				console.log(error);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/utils/log-buffer.ts:78]: This should be a `console.error()`?

File: packages/client/hmi-client/src/utils/logger.ts [hunk 7]
```diff
@@ -1,86 +1,210 @@
...
-					console.warn('POST to /logs did not return a 200');
-				} else {
-					this.clearLogs();
-				}
-			} catch (error) {
-				console.error(error);
-			}
+
+		if (this.options.consoleEnabled) {
+			console.log(`[${level}] ${message} ${this.callerInfo}`, ...optionalParams);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/utils/logger.ts:111]: Why not using the level for the console itself?  
```js
console.[level](`${message} ${this.callerInfo}`, ...optionalParams)
```

File: packages/client/hmi-client/src/utils/logger.ts [hunk 8]
```diff
@@ -1,86 +1,210 @@
...
+	INFO = 'info',
+	ERROR = 'error',
+	WARN = 'warn',
+	DEBUG = 'debug'
+}
+
+/**
+ * @interface ILoggerMessageOptions
+ */
+interface ILoggerMessageOptions {
```
Comment (YohannParis) [inline packages/client/hmi-client/src/utils/logger.ts:23]: Same comment as above about naming types [per documentation](https://github.com/DARPA-ASKEM/TERArium/blob/main/packages/client/hmi-client/DEVELOPMENT.md#types)

**Review by YohannParis — CHANGES_REQUESTED** [review]
Thanks, could you also update the documentation on how to use the Toast/Log system on the DEVELOPMENT.md.

**Comment by tangopapatime** [comment]
Thanks for the feedback @YohannParis .  Will look into addressing them!

File: packages/client/hmi-client/src/utils/log-buffer.ts (hunk 3 above)
Comment (tangopapatime) [reply packages/client/hmi-client/src/utils/log-buffer.ts:27]: Yeah.  I wasn't exactly sure of what the best approach was.  The one on `app.vue` re routes you to the login page.  The `api.vue` might not have access to this vue router yet when api is being used.  I'm all ears if there's a particular pattern that solves this.

File: packages/client/hmi-client/src/utils/log-buffer.ts (hunk 4 above)
Comment (tangopapatime) [reply packages/client/hmi-client/src/utils/log-buffer.ts:32]: Yeah I remember this but maybe my comment about thinking it's weird to force `Interface` usage and `Type` nomenclature lol.  Will change it to `Type` cause I don't have big feelings over it :)

File: packages/services/hmi-server/docker/jvm/Dockerfile.jvm [hunk 9]
```diff
@@ -1,7 +1,7 @@
 ####
 # This Dockerfile is used in order to build a container that runs the Quarkus application in JVM mode
 ###
-FROM registry.access.redhat.com/ubi8/openjdk-17:1.14-10
+FROM registry.access.redhat.com/ubi8/openjdk-17:1.14-10.1675788279
```
Comment (YohannParis) [inline packages/services/hmi-server/docker/jvm/Dockerfile.jvm:4]: Are we sure we want to do that.
I wonder if your branch is out of date with `main`?

File: packages/client/hmi-client/src/utils/logger.ts [hunk 10]
```diff
@@ -126,30 +126,30 @@ class Logger {
 	/**
 	 *
 	 * @param {string} level
-	 * @param {*} msg
+	 * @param {*} message
```
Comment (YohannParis) [inline packages/client/hmi-client/src/utils/logger.ts:129]: This is amazing!
I love long variables name

**Review by YohannParis — APPROVED** [review]

**Review by YohannParis — APPROVED** [review]

**Comment by tangopapatime** [comment]
I apologize for the extremely linty PR.

**Review by YohannParis — APPROVED** [review]

**Pull request merged**